### PR TITLE
Top-hat transform

### LINKED ISSUE #2341: Top-hat transform

### PR DESCRIPTION
**Pull request by 0x00b1**
Resolves #2341.

### CODE REVIEW

**Comment by dlogan** [comment]
Neat! So how is this different from EnhanceOrSuppressFeatures > Speckles?  Or Morph > tophat? (Very likely I may have missed some CP feature reorganization, so please bear with me)

**Comment by 0x00b1** [comment]
😎

`EnhanceOrSuppressFeatures` exposes a white top-hat transform, but not the equally useful black top-hat transform. This and a [few other modules](https://github.com/CellProfiler/CellProfiler/projects/5) will replace `Morph`.

(This is also faster and supports multi-dimensional images.)

File: cellprofiler/modules/tophattransform.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+"""
+
+<strong>Top-hat transform</strong>
```
Comment (mcquin) [inline cellprofiler/modules/tophattransform.py]: Documentation

File: cellprofiler/modules/tophattransform.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import skimage.morphology
+
+
+class TopHatTransform(cellprofiler.module.Module):
+    category = "Mathematical morphology"
```
Comment (mcquin) [inline cellprofiler/modules/tophattransform.py]: The mathematical morphology modules are all very similar. Perhaps we could create a `cellprofiler.module.MathematicalMorthpholoy` class which encompasses the repeated behavior.

File: cellprofiler/modules/tophattransform.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        self.x_name = cellprofiler.setting.ImageNameSubscriber(
+            "Input"
+        )
+
+        self.y_name = cellprofiler.setting.ImageNameProvider(
+            "Output",
+            "TopHatTransform"
+        )
+
+        self.operation_name = cellprofiler.setting.Choice(
```
Comment (mcquin) [inline cellprofiler/modules/tophattransform.py]: Documentation.

File: cellprofiler/modules/tophattransform.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            workspace.display_data.y_data = y_data
+
+            workspace.display_data.dimensions = dimensions
+
+    def display(self, workspace, figure):
+        figure.set_subplots((2, 1), dimensions=workspace.display_data.dimensions)
+
+        figure.subplot_imshow(0, 0, workspace.display_data.x_data, colormap="gray", dimensions=workspace.display_data.dimensions)
+
+        figure.subplot_imshow(1, 0, workspace.display_data.y_data, colormap="gray", dimensions=workspace.display_data.dimensions)
```
Comment (mcquin) [inline cellprofiler/modules/tophattransform.py]: Should the color map be non-gray for multi-channel images?

**Pull request merged**